conviva/src/main/java/com/bitmovin/analytics/conviva/ConvivaAnalyticsIntegration.java

```diff
@@ -230,95 +176,102 @@ public void reportPlaybackDeficiency(String message, Client.ErrorSeverity severi
      * Puts the session state in a notMonitored state.
      */
     public void pauseTracking() {
-        try {
-            client.adStart(sessionId, Client.AdStream.SEPARATE, Client.AdPlayer.SEPARATE, Client.AdPosition.PREROLL);
-            client.detachPlayer(sessionId);
-            Log.d(TAG, "Tracking paused.");
-        } catch (ConvivaException e) {
-            Log.e(TAG, e.getLocalizedMessage());
+        Log.d(TAG, "Will report ad break started: " + ConvivaSdkConstants.AdPlayer.SEPARATE + " " + ConvivaSdkConstants.AdType.CLIENT_SIDE);
+        convivaVideoAnalytics.reportAdBreakStarted(ConvivaSdkConstants.AdPlayer.SEPARATE, ConvivaSdkConstants.AdType.CLIENT_SIDE);
+        if(isSessionActive) {
+            convivaVideoAnalytics.reportPlaybackEnded();
+            isSessionActive = false;
         }
 
+        Log.d(TAG, "Tracking paused.");
     }
 
     /**
      * Puts the session state from a notMonitored state into the last one tracked.
      */
     public void resumeTracking() {
-        try {
-            client.attachPlayer(sessionId, playerStateManager);
-            client.adEnd(sessionId);
-            Log.d(TAG, "Tracking resumed.");
-        } catch (ConvivaException e) {
-            Log.e(TAG, e.getLocalizedMessage());
+        if(!isSessionActive) {
+            convivaVideoAnalytics.reportPlaybackRequested();
+            isSessionActive = true;
         }
+
+        convivaVideoAnalytics.reportAdBreakEnded();
+        Log.d(TAG, "Tracking resumed.");
+    }
+
+    /**
+     * This should be called when the app is resumed
+     */
+    public void reportAppForegrounded() {
+        Log.d(TAG, "appForegrounded");
+        ConvivaAnalytics.reportAppForegrounded();
```

Comment:
`isSessionActive` is set to `false` in reportAppBackgrounded but is not set to `true` again here. Is this expected? What about the use case when application is backgrounded thus marking session as inactive and then user foregrounds app resuming playback?

Reply:
I implemented it this way because it was what worked in touchstone, after some trial and error. But again, I have to research the app backgrounding issue.